go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -421,6 +404,75 @@ func (c *Controller) syncServices(key string) error {
 	return nil
 }
 
+func (c *Controller) addServiceToIdlingBalancer(vips sets.String, service *v1.Service) error {
+	for _, vipProtocol := range vips.List() {
+		vip, protocol := splitVirtualIPKey(vipProtocol)
+		lb, err := loadbalancer.GetOVNKubeIdlingLoadBalancer(protocol)
+		if err != nil {
+			return errors.Wrapf(err, "Error getting OVN idling LoadBalancer for protocol %s", protocol)
+		}
+		loadbalancer.UpdateLoadBalancer(lb, vip, []string{})
+		// update the tracker with the VIP
+		c.serviceTracker.updateService(service.Name, service.Namespace, vip, protocol)
+		// mark the vip as processed
+	}
+	return nil
+}
+
+// collectServiceVIPs collects all the vips associated to a given service
+// and returns them as a set.
+func (c *Controller) collectServiceVIPs(service *v1.Service) sets.String {
```

Comment:
why is this a method? it should be a function and maybe better in utils?

Reply:
Right, moved